### PR TITLE
[Bugfix] Fix non-contiguous a/b tensors in GDN when num_v_heads == num_k_heads

### PR DESCRIPTION
**Pull request by b0nce**
Fixes #21843

## Motivation

`Qwen3GatedDeltaNet` produces incorrect results when configured with `num_v_heads == num_k_heads` (ratio=1). Models with ratio=2 and ratio=4 work as expected.

In `fix_query_key_value_ordering`, after `torch.split` on `mixed_ba` and `reshape`, the `a` and `b` tensors end up non-contiguous with strides `(2*num_k_heads, 2)` instead of `(num_v_heads, 1)`. The `fused_gdn_gating` triton kernel indexes these as flat contiguous arrays, so it reads interleaved a/b values from the packed tensor.

This only affects ratio=1 because the reshape from `[S, num_k_heads, 1]` to `[S, num_v_heads]` merely squeezes a trailing dim, preserving the non-contiguous strides. For ratio>=2 the reshape genuinely changes the shape (e.g. `[S, 16, 2]` to `[S, 32]`), which forces a contiguous copy automatically.

Likely never caught because the original Qwen3 GDN uses ratio=2 (`num_v_heads = 2 * num_k_heads`), where the bug doesn't manifest.

## Modifications

Added `.contiguous()` to `a` and `b` after reshape in `fix_query_key_value_ordering` (`python/sglang/srt/models/qwen3_next.py`). No-op for ratio>=2 (already contiguous).

## Accuracy Tests

See `test/srt/cpu/test_gdn_contiguity.py`. Tests that `a` and `b` are contiguous and readable by triton kernels assuming flat layout, for ratio 1, 2, and 4.

## Benchmarking and Profiling

No performance impact. `.contiguous()` is a no-op when the tensor is already contiguous (ratio>=2). For ratio=1 it copies a small `[seq_len, num_v_heads]` tensor.

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

File: python/sglang/srt/models/qwen3_next.py [hunk 1]
```diff
@@ -324,8 +324,8 @@ def fix_query_key_value_ordering(
         # [b, sq, ng, np/ng * hn] -> [b, sq, np, hn]
         value = value.reshape(value.size(0), -1, self.head_v_dim)
         z = z.reshape(z.size(0), -1, self.head_v_dim)
-        b = b.reshape(b.size(0), self.num_v_heads // self.attn_tp_size)
-        a = a.reshape(a.size(0), self.num_v_heads // self.attn_tp_size)
+        b = b.reshape(b.size(0), self.num_v_heads // self.attn_tp_size).contiguous()
```
[the remark this answers is not in the document]
Comment (b0nce) [reply python/sglang/srt/models/qwen3_next.py:327]: `z` is non-contiguous after the split for ratio=1, but it goes through `z.reshape(-1, z.shape[-1])` at line 406 before being passed to `self.norm`. Since the tensor is non-contiguous, `reshape` creates a contiguous copy there, so the norm kernel always receives contiguous input. Similarly, `value` gets flattened and concatenated into `mixed_qkv` via `torch.cat`, which also produces a contiguous result. Adding `.contiguous()` here would just move the copy earlier rather than eliminate a bug.

File: tests/repro_gdn_ratio1_bug.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    ratio = num_v_heads // num_k_heads
+    nk_tp = num_k_heads // tp_size
+    nv_tp = num_v_heads // tp_size
+
+    new_shape = mixed_ba.size()[:-1] + (nk_tp, 2 * ratio)
+    mixed_ba = mixed_ba.view(*new_shape)
+    b, a = torch.split(mixed_ba, [ratio, ratio], dim=2)
+
+    b = b.reshape(b.size(0), nv_tp)
+    a = a.reshape(a.size(0), nv_tp)
```
[the remark this answers is not in the document]
Comment (b0nce) [reply tests/repro_gdn_ratio1_bug.py]: Addressed - the test now has both a buggy and fixed version of the function, asserts that the bug exists for ratio = 1 in the original code, and verifies the `.contiguous()` fix works for all ratios.

**Comment by cs-cat** [comment]
Hello @b0nce , you can test if PR #22312 resolves this issue. The GDN kernel now supports non-contiguous B/A tensors, which also fixes the accuracy issue in Qwen3.5-27B.
